go/test/endtoend/recovery/unshardedrecovery/recovery.go

```diff
@@ -0,0 +1,332 @@
+	assert.Nil(t, err)
+	cluster.VerifyRowsInTablet(t, replica1, keyspaceName, 3)
+
+	recovery.RestoreTablet(t, localCluster, replica3, recoveryKS2, "0", keyspaceName, commonTabletArg)
+
+	output, err = localCluster.VtctlclientProcess.ExecuteCommandWithOutput("GetVSchema", recoveryKS2)
+	assert.Nil(t, err)
+	assert.Contains(t, output, "vt_insert_test")
+
+	cluster.VerifyRowsInTablet(t, replica3, keyspaceName, 2)
+
+	// update the original row in master
+	_, err = master.VttabletProcess.QueryTablet("update vt_insert_test set msg = 'msgx2' where id = 1", keyspaceName, true)
+	assert.Nil(t, err)
+
+	//verify that master has new value
+	qr, err = master.VttabletProcess.QueryTablet("select msg from vt_insert_test where id = 1", keyspaceName, true)
+	assert.Nil(t, err)
+	assert.Equal(t, "msgx2", fmt.Sprintf("%s", qr.Rows[0][0].ToBytes()))
+
+	//verify that restored replica has old value
+	qr, err = replica3.VttabletProcess.QueryTablet("select msg from vt_insert_test where id = 1", keyspaceName, true)
+	assert.Nil(t, err)
+	assert.Equal(t, "msgx1", fmt.Sprintf("%s", qr.Rows[0][0].ToBytes()))
+
+	vtgateInstance := localCluster.GetVtgateInstance()
+	vtgateInstance.TabletTypesToWait = "REPLICA"
+	err = vtgateInstance.Setup()
+	localCluster.VtgateGrpcPort = vtgateInstance.GrpcPort
+	assert.Nil(t, err)
+	err = vtgateInstance.WaitForStatusOfTabletInShard(fmt.Sprintf("%s.%s.master", keyspaceName, shardName), 1)
+	assert.Nil(t, err)
+	err = vtgateInstance.WaitForStatusOfTabletInShard(fmt.Sprintf("%s.%s.replica", keyspaceName, shardName), 1)
+	assert.Nil(t, err)
+	err = vtgateInstance.WaitForStatusOfTabletInShard(fmt.Sprintf("%s.%s.replica", recoveryKS1, shardName), 1)
+	assert.Nil(t, err)
+	err = vtgateInstance.WaitForStatusOfTabletInShard(fmt.Sprintf("%s.%s.replica", recoveryKS2, shardName), 1)
+	assert.Nil(t, err)
+
+	// Build vtgate grpc connection
+	grpcAddress := fmt.Sprintf("%s:%d", localCluster.Hostname, localCluster.VtgateGrpcPort)
+	vtgateConn, err := vtgateconn.Dial(context.Background(), grpcAddress)
+	assert.Nil(t, err)
+	session := vtgateConn.Session("@replica", nil)
+
+	//check that vtgate doesn't route queries to new tablet
+	recovery.VerifyQueriesUsingVtgate(t, session, "select count(*) from vt_insert_test", "INT64(3)")
+	recovery.VerifyQueriesUsingVtgate(t, session, "select msg from vt_insert_test where id = 1", `VARCHAR("msgx2")`)
+	recovery.VerifyQueriesUsingVtgate(t, session, fmt.Sprintf("select count(*) from %s.vt_insert_test", recoveryKS1), "INT64(1)")
+	recovery.VerifyQueriesUsingVtgate(t, session, fmt.Sprintf("select msg from %s.vt_insert_test where id = 1", recoveryKS1), `VARCHAR("test1")`)
+	recovery.VerifyQueriesUsingVtgate(t, session, fmt.Sprintf("select count(*) from %s.vt_insert_test", recoveryKS2), "INT64(2)")
+	recovery.VerifyQueriesUsingVtgate(t, session, fmt.Sprintf("select msg from %s.vt_insert_test where id = 1", recoveryKS2), `VARCHAR("msgx1")`)
+
```

Comment:
We also need to check that `use keyspace@replica` and `use keyspace:shard@replica` work correctly in this test, similar to how it has been done in sharded_recovery test and the original python test.
https://github.com/planetscale/vitess/blob/master/go/test/endtoend/recovery/shardedrecovery/sharded_recovery_test.go#L458
https://github.com/planetscale/vitess/blob/master/test/recovery.py#L327